xaitk_saliency/impls/vis_sal_classifier/occlusion_scoring.py

```diff
@@ -0,0 +1,61 @@
+from xaitk_saliency import ImageClassifierSaliencyMapGenerator
+from xaitk_saliency.utils.masking import weight_regions_by_scalar
+
+import numpy as np
+import warnings
+
+
+class OcclusionScoring (ImageClassifierSaliencyMapGenerator):
+    """
+    This saliency implementation transforms black-box
+    image classification scores into saliency heatmaps. This should
+    require a sequence of per-class confidences predicted on the
+    reference image, a number of per-class confidences as predicted
+    on perturbed images, as well as the masks of the reference image
+    perturbations (as would be output from a
+    `PerturbImage` implementation.
+
+    The perturbation masks used by the following implementation are
+    expected to be of type integer. Masks containing values of type
+    float are rounded to the nearest value and binarized
+    with value 1 replacing values greater than or equal to half of
+    the maximum value in mask after rounding while 0 replaces the rest.
+    """
+
+    def generate(
+            self,
+            image_conf: np.ndarray,
+            perturbed_conf: np.ndarray,
+            perturbed_masks: np.ndarray
+    ) -> np.ndarray:
+
+        if isinstance(perturbed_masks[0][0][0], float):
+            # Roundoff when perturbation mask value is of type float
+            perturbed_masks = np.round(perturbed_masks)
+            # Compute binary midpoint for setting threshold
+            binary_midpoint = (np.max(perturbed_masks))/2
+            # Binarizing mask values to be of type int and range [0, 1]
+            perturbed_masks = np.where(perturbed_masks > binary_midpoint,
```

Comment:
I think this line is equivalent to `perturbed_masks = perturbed_masks > binary_midpoint`, which results in a boolean-typed matrix.

Reply:
You are correct! I changed this to the simpler form suggested.